src/components/Graph/index.tsx

```diff
@@ -0,0 +1,254 @@
+  const label = dataset.label
+  let data: Point[] = []
+  let backgroundColor: string[] = []
+  let borderColor: string[] = []
+
+  dataset.data.forEach(d => {
+    data.push(d.y)
+
+    const background = d.backgroundColor ? d.backgroundColor : dataset.backgroundColor;
+    const border = d.borderColor ? d.borderColor : dataset.borderColor;
+    if(background !== undefined) {
+      backgroundColor.push(background)
+    }
+
+    if(border !== undefined) {
+      borderColor.push(border)
+    }
+  })
+
+  let pointBackgroundColor = backgroundColor
+  let pointBorderColor = borderColor
+
+  return {
+    label,
+    data,
+    backgroundColor,
+    borderColor,
+    pointBackgroundColor,
+    pointBorderColor,
+    borderWidth: 1,
+  }
+}
+
+function getChartDatasets(datasets: Dataset[]): ChartDataSets[] {
+  return datasets.map(getChartDataset)
+}
+
+function getLabelsFromDataset(datasets: Dataset[]) {
+  let labels: string[] = []
+  datasets.forEach(dataset =>
+    dataset.data.forEach(d => {
+      if (!labels.includes(d.x)) {
+        labels.push(d.x)
+      }
+    }),
+  )
+
+  return labels
+}
+
+function getChartData(graph: Graph): ChartData {
+  return {
+    labels: getLabelsFromDataset(graph.datasets),
+    datasets: getChartDatasets(graph.datasets),
+  }
+}
+
+function getCommonChartOptions(graph: Graph): ChartOptions {
+  const options: ChartOptions = {
+    title: {
+      display: !!graph.title,
+      text: graph.title,
+    },
+  }
+
+  if (options.title) {
+    if (graph.titleFontSize) {
+      options.title.fontSize = graph.titleFontSize
+    }
+
+    if (graph.titleFontColor) {
+      options.title.fontColor = graph.titleFontColor
+    }
+  }
+
+  return options
+}
+
+function getCommonChartConfigurations(type: string, graph: Graph): ChartConfiguration {
+  return {
+    type,
+    data: getChartData(graph),
+    options: getCommonChartOptions(graph),
+  }
+}
+
+interface IBarGraph extends Graph, HasAxes {
+  /** Determines if the bar graph should be displayed in a horizontal manner */
+  horizontal?: boolean
+  /** Determines if the bar graph should be displayed as a stacked bar graph */
+  stacked?: boolean
+}
+
+class BarGraph extends Component<IBarGraph, {}> {
+  graph: any
+  chart: any
+
+  componentDidMount() {
+    const stacked = !!this.props.stacked;
+    const { horizontal, xAxes, yAxes } = this.props
+    const type = horizontal ? 'horizontalBar' : 'bar'
+    const config = getCommonChartConfigurations(type, this.props)
+
+    if (config && config.options) {
+      let scales = undefined
+      if (!horizontal) {
+        scales = {
+          xAxes: getAxes(xAxes, stacked),
+          yAxes: getAxes(yAxes, stacked),
+        }
+      } else {
+        scales = {
+          xAxes: getAxes(yAxes, stacked),
+          yAxes: getAxes(xAxes, stacked),
+        }
+      }
+
+      config.options.scales = scales
+    }
+
+    this.graph = new ChartJs(this.chart, config)
```

Comment:
I don't understand how you use `this.chart`: where you initialize it?

Reply:
this.chart is used in the render method. 

The `ChartJs`requires an element to be passed in as the first argument to the constructor. this.chart is the element where the Chart should be rendered to. I've updated the type of `this.chart` to be `HTMLCanvasElement` so hopefully the code is a little more clear. 

This methodology is similar to how [react-chartjs-2](https://www.npmjs.com/package/react-chartjs-2#chartjs-instance) does it